### ISSUE TITLE
[Failing Test] gci-gce-alpha-enabled-default

### ISSUE THREAD

**Issue by pacoxu** [kubernetes/kubernetes#128600]
### Which jobs are failing?

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-enabled-default/1854054086446419968

### Which tests are failing?

- Kubernetes e2e suite.[It] [sig-api-machinery] API priority and fairness should ensure that requests can be classified by adding FlowSchema and PriorityLevelConfiguration[Changes](https://github.com/kubernetes/kubernetes/compare/9a2a7537f...a28f14089?)
- Kubernetes e2e suite.[It] [sig-api-machinery] ValidatingAdmissionPolicy [Privileged:ClusterAdmin] should type check a CRD[Changes](https://github.com/kubernetes/kubernetes/compare/9a2a7537f...a28f14089?)
- Kubernetes e2e suite.[It] [sig-api-machinery] ValidatingAdmissionPolicy [Privileged:ClusterAdmin] should type check validation expressions[Changes](https://github.com/kubernetes/kubernetes/compare/9a2a7537f...a28f14089?)Kubernetes e2e suite.[It] [sig-auth] ValidatingAdmissionPolicy can restrict access by-node[Changes](https://github.com/kubernetes/kubernetes/compare/9a2a7537f...a28f14089?)
- Kubernetes e2e suite.[It] [sig-cli] Kubectl client Simple pod should return command exit codes should handle in-cluster config[Changes](https://github.com/kubernetes/kubernetes/compare/9a2a7537f...a28f14089?)

### Since when has it been failing?

11-06

### Testgrid link

https://testgrid.k8s.io/google-gce#gci-gce-alpha-enabled-default

### Reason for failure (if possible)

https://github.com/kubernetes/kubernetes/compare/9a2a7537f...a28f14089 
 
~~potential root cause:~~
- ~~#126605~~
- ~~#126806~~
- ~~#127404~~
- ~~#128377~~
- ~~#128539~~ 

https://github.com/kubernetes/kubernetes/pull/128266/ according to https://github.com/kubernetes/kubernetes/issues/128600#issuecomment-2459923403 @aojea investigation.

**Comment by aojea** [comment]
nice, we have 5 potential root causes :) 

It seems all tests failing are timing out :/

**Comment by aojea** [comment]
/remove-sig cloud-provider
/SIG api-machinery
/assign @benluddy 
/cc @liggitt @deads2k 

This seems the culprit https://github.com/kubernetes/kubernetes/pull/128539, if all alphas are enabled it seems to break some flows 


Checking to this job failure https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-enabled-default/1854054086446419968

API server logs https://storage.googleapis.com/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-enabled-default/1854054086446419968/artifacts/bootstrap-e2e-master/kube-apiserver.log
```
  }
E1106 06:57:52.632897      10 status.go:71] "Unhandled Error" err="apiserver received an error that is not an metav1.Status: &errors.errorString{s:\"no serializer defined for application/cbor\"}: no serializer defined for application/cbor" logger="UnhandledError"
I1106 06:57:52.634673      10 httplog.go:134] "HTTP" verb="PATCH" URI="/apis/flowcontrol.apiserver.k8s.io/v1/flowschemas/system-nodes/status?fieldManager=api-priority-and-fairness-config-consumer-v1&force=true" latency="2.153195ms" userAgent="kube-apiserver/v1.32.0 (linux/amd64) kubernetes/b631dae" audit-ID="314ad10b-f8d4-423f-add2-b0196c3503b6" srcIP="[::1]:38828" apf_pl="exempt" apf_fs="exempt" apf_iseats=1 apf_fseats=0 apf_additionalLatency="0s" apf_execution_time="1.877089ms" resp=500 statusStack=<
	
	goroutine 6837 [running]:
	k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc0087dc210, 0xc0087ee3a8?)
		k8s.io/apiserver/pkg/server/httplog/httplog.go:336 +0xf2
	k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc0087dc210, 0x1f4)
		k8s.io/apiserver/pkg/server/httplog/httplog.go:316 +0x1d
	k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc0087e2a80, 0x1f4)
		k8s.io/apiserver/pkg/server/filters/timeout.go:237 +0x1cc
	k8s.io/apiserver/pkg/endpoints/filters.(*auditResponseWriter).WriteHeader(0xc0087f0060, 0x1f4)
		k8s.io/apiserver/pkg/endpoints/filters/audit.go:253 +0x34
	k8s.io/apiserver/pkg/endpoints/metrics.(*ResponseWriterDelegator).WriteHeader(0x2fe4840?, 0xc0087ee2c0?)
		k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:810 +0x26
	k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.(*deferredResponseWriter).Write(0xc0087f06c0, {0xc008783730, 0x5a, 0x61})
		k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:244 +0x669
	k8s.io/apimachinery/pkg/runtime/serializer/protobuf.(*Serializer).doEncode(0xc0002420c0, {0x3ca4ca8, 0xc0087ab2c0}, {0x3c9a0c0, 0xc0087f06c0}, {0x3c99f80?, 0x5b911a0?})
		k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go:228 +0x868
	k8s.io/apimachinery/pkg/runtime/serializer/protobuf.(*Serializer).encode(0xc0002420c0, {0x3ca4ca8?, 0xc0087ab2c0}, {0x3c9a0c0, 0xc0087f06c0}, {0x3c99f80, 0x5b911a0})
		k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go:181 +0x105
	k8s.io/apimachinery/pkg/runtime/serializer/protobuf.(*Serializer).Encode(0x0?, {0x3ca4ca8?, 0xc0087ab2c0?}, {0x3c9a0c0?, 0xc0087f06c0?})
		k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go:174 +0x35
	k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*codec).doEncode(0xc0087ab400, {0x3ca4ca8, 0xc0087ab2c0}, {0x3c9a0c0, 0xc0087f06c0}, {0x0?, 0x0?})
		k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go:268 +0x4fe
	k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*codec).encode(0xc0087ab400, {0x3ca4ca8?, 0xc0087ab2c0}, {0x3c9a0c0, 0xc0087f06c0}, {0x0, 0x0})
		k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go:214 +0x119
	k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*codec).Encode(0x3cd47c0?, {0x3ca4ca8?, 0xc0087ab2c0?}, {0x3c9a0c0?, 0xc0087f06c0?})
		k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go:207 +0x2d
	k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.SerializeObject({0x36c0a1f, 0x23}, {0x7bae124fa238, 0xc0087ab400}, {0x3cc7410, 0xc0087da700}, 0xc0087ecf00, 0x1f4, {0x3ca4ca8, 0xc0087ab2c0})
		k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:112 +0xa75
	k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated.func2()
		k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:298 +0x1fd
	k8s.io/apiserver/pkg/endpoints/request.(*durationTracker).Track(0xc0087e2870, 0xc0087e6d00)
		k8s.io/apiserver/pkg/endpoints/request/webhook_duration.go:75 +0x76
	k8s.io/apiserver/pkg/endpoints/request.TrackSerializeResponseObjectLatency({0x3cd47c0?, 0xc0087e2ea0?}, 0xc0087e6d00)
		k8s.io/apiserver/pkg/endpoints/request/webhook_durat
```

**Comment by benluddy** [comment]
There's a PR queued up for this: https://github.com/kubernetes/kubernetes/pull/128503.

**Comment by aojea** [comment]
> There's a PR queued up for this: #128503.

perfect, can you add to the descriptiion the `Fixes: ` thing so the bot autocloses this once that merges?

**Comment by liggitt** [comment]
weirdly, pull-kubernetes-e2e-gce-cos-alpha-features passed on https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128539/pull-kubernetes-e2e-gce-cos-alpha-features/1853892533789134848 ... but then started failing? I'm very confused how that happened

**Comment by benluddy** [comment]
> I'm very confused how that happened

Me too, we knew that the second PR was going to be necessary to serve built-in APIs to CBOR clients, and that the client-go feature gate that causes the dynamic client to enable CBOR would be turned on by AllAlpha, but the presubmit was green.

Those stack traces are from HandleError but the effect should just be a 500 response to clients that opted in to CBOR.

**Comment by benluddy** [comment]
> even start a server

Most of the tests are passing in the job I am looking at, where is the server failing to start?

**Comment by aojea** [comment]
my bad then, sorry, I correlated too fast then :/

The periodic seems to have a trigger

![image](https://github.com/user-attachments/assets/cfbcf134-935a-46e0-9027-daabd20f3843)

The presubmit is totally random 
https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-cos-alpha-features

**Comment by benluddy** [comment]
Oh, does the presubmit -focus only those tests tagged for alpha features and the periodic does not? https://github.com/kubernetes/test-infra/blob/9399ae78380ba034840fdf2f3e041c0ddd9b1114/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L957

**Comment by aojea** [comment]
I don't know is related by it is in the other alpha job that started to fail a bit later https://testgrid.k8s.io/google-gce#gce-cos-master-alpha-features

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-features/1854070444634673152

```
inkgo detected an issue with your spec structure
set up framework | framework.go:200
  It looks like you are trying to add a [BeforeEach] node
  to the Ginkgo spec tree in a leaf node after the specs started running.
  To enable randomization and parallelization Ginkgo requires the spec tree
  to be fully constructed up front.  In practice, this means that you can
  only create nodes like [BeforeEach] at the top-level or within the
  body of a Describe, Context, or When.
  Learn more at:
  http://onsi.github.io/ginkgo/#mental-model-how-ginkgo-traverses-the-spec-hierarchy
Ginkgo ran 1 suite in 8m6.797868352s
```
cc: @pohly and he is the more familiar with the ginkgo hooks logic in the e2e framework

**Comment by aojea** [comment]
The time 

![image](https://github.com/user-attachments/assets/39268f8b-397b-4de4-8b26-751c15cbe513)


points to this one https://github.com/kubernetes/kubernetes/pull/128266 :/

**Comment by aojea** [comment]
https://github.com/kubernetes/kubernetes/pull/128266/files#r1831130829

**Comment by liggitt** [comment]
> Oh, does the presubmit -focus only those tests tagged for alpha features and the periodic does not? https://github.com/kubernetes/test-infra/blob/9399ae78380ba034840fdf2f3e041c0ddd9b1114/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L957

oh... yeah, seems so 🤦

**Comment by benluddy** [comment]
The ginkgo issue looks distinct to me. I'm not surprised that the tests in this issue's description could be broken by the alpha gate that controls the dynamic client encoding, and that the presubmit simply doesn't run any of them.

**Comment by aojea** [comment]
ok, we have to figure this out, I think there are multiple things going on and the alpha feature enablement seems to be the one exposing them

**Comment by aojea** [comment]
ok, we are one down,https://github.com/kubernetes/kubernetes/issues/128607#issuecomment-2460029204 , let's try to understand this now

**Comment by benluddy** [comment]
> let's try to understand this now

The original explanation still looks correct to me. This was intended to be addressed by the second PR, it was just not clear that these tests would be affected because the alpha presubmit job does not include these tests.

**Comment by benluddy** [comment]
Result for https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128503/pull-kubernetes-e2e-kind-alpha-features/1854141845752254464 came in green -- it's a different alpha job, but it does run all of the tests failing in this issue so that is encouraging. Still waiting on results for https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128503/pull-e2e-gci-gce-alpha-enabled-default/1854186795659431936.

**Comment by benluddy** [comment]
Got results the first run of this job post-merge (https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-e2e-gci-gce-alpha-enabled-default/1854326383216431104). All of the tests that had been failing are green. There are 3 new storage failures that don't seem related.

Looking into the first one, `[sig-storage] In-tree Volumes [Driver: local] [LocalVolumeType: blockfs] [Testpattern: Pre-provisioned PV (filesystem volmode)] volumeMode [It] should not mount / map unused volumes in a pod [LinuxOnly] [sig-storage]` fails to connect from the apiserver to a kubelet:

```
I1107 01:09:08.484315 11000 volumemode.go:400] Unexpected error: error checking directory /var/lib/kubelet/pods/778314c4-c487-454f-a17f-02c49fd0c448/volumeDevices on node bootstrap-e2e-minion-group-h288: error dialing backend: dial tcp 10.40.0.4:10250: connect: connection refused
```

and the cause seems to be that this kubelet had just crashed:

```
Nov 07 01:09:07.338827 bootstrap-e2e-minion-group-h288 kubelet[1937]: fatal error: concurrent map writes
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]: goroutine 29497 [running]:
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet/cm/containermap.ContainerMap.Add(...)
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/cm/containermap/container_map.go:36
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet/cm/cpumanager.(*manager).AddContainer(0xc0008a5ad0, 0xc002175688, 0xc002d8a780, {0xc002d21ec0, 0x40})
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/cpu_manager.go:276 +0x165
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet/cm.(*internalContainerLifecycleImpl).PreStartContainer(0xc000a2a060, 0xc002175688, 0xc002d8a780, {0xc002d21ec0, 0x40})
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/cm/internal_container_lifecycle.go:42 +0x42
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet/kuberuntime.(*kubeGenericRuntimeManager).startContainer(0xc0009c4180, {0x334ff38, 0xc001184050}, {0xc0017ab600, 0x40}, 0xc000323420, 0xc0026aad58, 0xc002175688, 0xc002856c60, {0x4de4ae0, ...}, ...)
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_container.go:271 +0xcdb
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet/kuberuntime.(*kubeGenericRuntimeManager).SyncPod.func1({0x334ff38, 0xc001184050}, {0x2e91ac3, 0x9}, {0x2e91ac3, 0x9}, 0xc0026aad58)
Nov 07 01:09:07.342609 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager.go:1249 +0x8c7
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet/kuberuntime.(*kubeGenericRuntimeManager).SyncPod(0xc0009c4180, {0x334ff38, 0xc001184050}, 0xc002175688, 0xc002856c60, {0x4de4ae0, 0x0, 0x0}, 0xc00095f220)
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager.go:1318 +0x37d9
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).SyncPod(0xc000431008, {0x334ff00, 0xc001133860}, 0x2, 0xc002175688, 0x0, 0xc002856c60)
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/kubelet.go:2000 +0x1923
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).podWorkerLoop.func1({0x0, {0x2, {0xc1c322abaa0c010a, 0x7267b7d9ca, 0x4dc1f60}, 0xc002175688, 0x0, 0x0, 0x0}}, 0xc000ad6960, ...)
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/pod_workers.go:1286 +0x1ca
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).podWorkerLoop(0xc000ad6960, {0xc000e53bc0, 0x24}, 0xc002574150)
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/pod_workers.go:1291 +0x49b
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).UpdatePod.func1()
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/pod_workers.go:950 +0x118
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]: created by k8s.io/kubernetes/pkg/kubelet.(*podWorkers).UpdatePod in goroutine 250
Nov 07 01:09:07.344799 bootstrap-e2e-minion-group-h288 kubelet[1937]:         k8s.io/kubernetes/pkg/kubelet/pod_workers.go:945 +0x20db
```

The second, `[sig-storage] CSI Mock volume fsgroup policies CSI FSGroupPolicy Update [LinuxOnly] [It] should update fsGroup if update from File to default [sig-storage] `, has a similar error at the same time:

```
I1107 01:09:08.341660 11112 csi_fsgroup_policy.go:175] Unexpected error: failed: writing the contents: error dialing backend: dial tcp 10.40.0.4:10250: connect: connection refused: 
```

And the same for the third failure, `[sig-storage] In-tree Volumes [Driver: local] [LocalVolumeType: blockfs] [Testpattern: Pre-provisioned PV (filesystem volmode)] volumeMode [It] should not mount / map unused volumes in a pod [LinuxOnly] [sig-storage]`:

```
I1107 01:09:08.484315 11000 volumemode.go:400] Unexpected error: error checking directory /var/lib/kubelet/pods/778314c4-c487-454f-a17f-02c49fd0c448/volumeDevices on node bootstrap-e2e-minion-group-h288: error dialing backend: dial tcp 10.40.0.4:10250: connect: connection refused 
```

I don't see an existing issue for this kubelet crash.

**Comment by benluddy** [comment]
OK, the following run was fully green. I think this is fixed. I'll open a new issue for the kubelet crash.
